### PR TITLE
Moved fs extra to core

### PR DESCRIPTION
**Pull request by akosyakov**

### CODE REVIEW

**Comment by akosyakov** [comment]
Could someone review?

File: packages/core/src/common/messaging/proxy-factory.ts [hunk 1]
```diff
@@ -143,17 +143,18 @@ export class JsonRpcProxyFactory<T extends object> implements ProxyHandler<T> {
...
-    protected onRequest(method: string, ...args: any[]): Promise<any> {
-        return new Promise<any>((resolve, reject) => {
-            try {
-                const promise = this.target[method](...args) as Promise<any>;
-                promise
-                    .catch(err => reject(err))
-                    .then(result => resolve(result));
-            } catch (err) {
-                reject(err);
+    protected async onRequest(method: string, ...args: any[]): Promise<any> {
```
Comment (kittaakos) [inline packages/core/src/common/messaging/proxy-factory.ts]: Cool! I am glad you have fixed this. We will receive real errors on the frontend with this change, right?

**Comment by kittaakos** [comment]
> Could someone review?

Does this fix the docker issue or you do not know unless it gets merged into the master? I can try to build the application from your branch but what else should be verified?

**Comment by akosyakov** [comment]
I don't know till it is merged. The issue appears when the app is built from published extensions. The local setup is quite different.

File: packages/core/src/common/messaging/proxy-factory.ts (hunk 1 above)
Comment (akosyakov) [reply packages/core/src/common/messaging/proxy-factory.ts]: yes

**Comment by akosyakov** [comment]
@hexa00 it should be fine

**Comment by akosyakov** [comment]
@kittaakos sorry, i've tested it locally it was wokring

**Comment by hexa00** [comment]
Note we should create an more general issue about this if there isn't one already so that we work on the underlying issue..?

**Pull request merged**